docs/src/main/tut/typeclasses/semigroup.md

```diff
@@ -7,92 +7,164 @@ scaladoc: "#cats.kernel.Semigroup"
+# Example usage: Merging maps
+
+Consider a function that merges two `Map`s that combines values if they share
+the same key. It is straightforward to write these for `Map`s with values of
+type say, `Int` or `List[String]`, but we can write it once and for all for
+any type with a `Semigroup` instance.
+
+```tut:book:silent
+import cats.instances.all._
+import cats.syntax.semigroup._
+
+def optionCombine[A: Semigroup](a: A, opt: Option[A]): A =
+  opt.map(a |+| _).getOrElse(a)
+
+def mergeMap[K, V: Semigroup](lhs: Map[K, V], rhs: Map[K, V]): Map[K, V] =
+  lhs.foldLeft(rhs) {
+    case (acc, (k, v)) => acc.updated(k, optionCombine(v, acc.get(k)))
+  }
+```
 
 ```tut:book
-Map("foo" -> Map("bar" -> 5)).combine(Map("foo" -> Map("bar" -> 6), "baz" -> Map()))
-Map("foo" -> List(1, 2)).combine(Map("foo" -> List(3,4), "bar" -> List(42)))
+val xm1 = Map('a' -> 1, 'b' -> 2)
+val xm2 = Map('b' -> 3, 'c' -> 4)
+
+val x = mergeMap(xm1, xm2)
+
+val ym1 = Map(1 -> List("hello"))
+val ym2 = Map(2 -> List("cats"), 1 -> List("world"))
+
+val y = mergeMap(ym1, ym2)
 ```
 
-which is far more likely to be useful than
+It is interesting to note that the type of `mergeMap` satisfies the type of `Semigroup`
+specialized to `Map[K, ?]` and is associative - indeed the `Semigroup` instance for `Map`
+uses the same function for its `combine`.
 
 ```tut:book
-Map("foo" -> Map("bar" -> 5)) ++  Map("foo" -> Map("bar" -> 6), "baz" -> Map())
-Map("foo" -> List(1, 2)) ++ Map("foo" -> List(3,4), "bar" -> List(42))
+Semigroup[Map[Char, Int]].combine(xm1, xm2) == x
+
+Semigroup[Map[Int, List[String]]].combine(ym1, ym2) == y
 ```
 
-There is inline syntax available for `Semigroup`. Here we are 
-following the convention from scalaz, that `|+|` is the 
-operator from `Semigroup`.
+# Exploiting laws: associativity
+
+Since we know `Semigroup#combine` must be associative, we can exploit this when writing
+code against `Semigroup`. For instance, to sum a `List[Int]` we can choose to either
+`foldLeft` or `foldRight` since all that changes is associativity.
 
-```tut:silent
-import cats.implicits._
+```tut:book
+val leftwards = List(1, 2, 3).foldLeft(0)(_ |+| _)
 
-val one = Option(1)
-val two = Option(2)
-val n: Option[Int] = None
+val rightwards = List(1, 2, 3).foldRight(0)(_ |+| _)
 ```
 
-Thus.
+Associativity also allows us to split a list apart and sum the parts in parallel, gathering the results in
+the end.
+
+```tut:book:silent
+val list = List(1, 2, 3, 4, 5)
+val (left, right) = list.splitAt(2)
+```
 
 ```tut:book
-one |+| two
-n |+| two
-n |+| n
-two |+| n
+val sumLeft = left.foldLeft(0)(_ |+| _)
+val sumRight = right.foldLeft(0)(_ |+| _)
+val result = sumLeft |+| sumRight
 ```
 
-You'll notice that instead of declaring `one` as `Some(1)`, I chose
-`Option(1)`, and I added an explicit type declaration for `n`. This is
-because there aren't type class instances for `Some` or `None`, but for
-`Option`. If we try to use `Some` and `None`, we'll get errors:
```

Comment:
I think we're going to move these "gotchas" to a more generic location, being handled at this PR: https://github.com/typelevel/cats/pull/1404